content/terms_and_definitions.md

```diff
@@ -1,285 +1,346 @@
+
+EXAMPLE   A root concept representing a wall, air outlet, or construction task is the root of a graph of connected entities and attributes, representing e.g., geometry, material, or breakdown structure.
+
+### concept template
+unit of functionality
+
+collection of objects and their relationships that defines one concept such that removal of any component would render the concept incomplete or ambiguous
+
+EXAMPLE  An example of a concept template in the context of a concept root, such as a wall, would consist of the collection of all entities and attributes required to specify a material definition independently of how it is utilized later.
+
+[SOURCE: ISO 10303-1:2021, 3.1.61, modified - unit of functionality used as admitted term, replaced "application objects and their relationships" by "entities and attributes", words "within the application context" removed, Example added]
+
+### concept usage
+
+use of a concept template in context of a concept root
+
+Note 1 to entry:  The use of a concept template may include applying constraints to its usage.
+
+EXAMPLE  The use of a concept template for material definitions for a particular concept root representing a wall.
 
 ## constraint
 
-restriction for a specified reason
+adaptation of a data type that restricts its range or operations
 
-> NOTE A specialization of the general term control.
+[SOURCE: ISO 2382:2015, 2122397]
 
 ## control
 
-directive to meet specified requirements such as for scope, time, or cost
+regulation of variables within specified limits
 
-> NOTE A specialization of the general term object.
+Note 1 to entry: Variables include scope, time and cost
 
-## dictionary
+## data dictionary
 
-collection of words, terms or concepts, with their definition
+database that contains metadata
 
-## element
+[SOURCE: ISO 12006-3:2022, 3.1]
 
-tangible physical product that can be described by its shape representation, material representations, and other properties
+## data types
 
-> NOTE A specialization of the general term product.
+### data type
 
-### element occurrence
+domain of values
 
-element's position within the project coordinate system and its containment within the spatial structure
+[SOURCE: ISO 10303-11:2004, 3.3.5, modified - The word "a" was removed.]
 
-## entity
+### enumeration data type
 
-class of information defined by common attributes and constraints as defined in ISO 10303-11
+data type having as its domain a set of names
 
-> NOTE Similar to the term "class" in common programming languages but describing data structure only (not behavior such as methods).
+[SOURCE: ISO 10302-11:2004, 8.4.1, modified - Definition adapted from normative text]
 
-## external reference
+### select data type
 
-link to information outside the data set, with direct relevance to the specific information the link originates from inside the data set
+data type that enables a choice among several named data types
```

Comment:
In the previous definition we have "one of multiple ...". Currently, "choice" isn't so clear that you can pick only one. Also I would slide in the word "union" which is the typical term to refer to these creatures.

```suggestion
data type that establishes the union of several named data types to allow an attribute value or aggregate element to be one of multiple types or entities
```

Needs some tweaking still